src/System.IO.FileSystem/tests/File/Copy_str_str.cs

```diff
@@ -1,439 +1,150 @@
-
-            strLoc = "Loc_498yg";
-
-
-
-            File.Create(filName).Dispose();
-            fil2 = new FileInfo(filName);
-            iCountTestcases++;
-            try
-            {
-                File.Copy(null, fil2.FullName);
-                iCountErrors++;
-                printerr("Error_209uz! Expected exception not thrown, fil2==" + fil1.FullName);
-                fil1.Delete();
-            }
-            catch (ArgumentNullException)
-            {
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_21x99! Incorrect exception thrown, exc==" + exc.ToString());
-            }
-            fil2.Delete();
+            File.Copy(source, dest);
+        }
 
-            // [] ArgumentNullException if null is passed in for destination file
+        #endregion
 
-            strLoc = "Loc_898gc";
+        #region UniversalTests
 
-            File.Create(filName).Dispose();
-            fil2 = new FileInfo(filName);
-            iCountTestcases++;
-            try
-            {
-                File.Copy(fil2.FullName, null);
-                iCountErrors++;
-                printerr("Error_8yt85! Expected exception not thrown");
-            }
-            catch (ArgumentNullException)
-            {
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_1298s! Incorrect exception thrown, exc==" + exc.ToString());
-            }
-
-
-            // [] String.Empty should throw ArgumentException
+        [Fact]
+        public void NullFileName()
+        {
+            Assert.Throws<ArgumentNullException>(() => Copy(null, "."));
+            Assert.Throws<ArgumentNullException>(() => Copy(".", null));
+        }
 
-            strLoc = "Loc_298vy";
+        [Fact]
+        public void EmptyFileName()
+        {
+            Assert.Throws<ArgumentException>(() => Copy(string.Empty, "."));
+            Assert.Throws<ArgumentException>(() => Copy(".", string.Empty));
+        }
 
-            File.Create(filName).Dispose();
-            fil2 = new FileInfo(filName);
-            iCountTestcases++;
-            try
-            {
-                File.Copy(fil2.FullName, String.Empty);
-                iCountErrors++;
-                printerr("Error_092u9! Expected exception not thrown, fil2==" + fil1.FullName);
-            }
-            catch (ArgumentException)
-            {
-            }
-            catch (Exception exc)
+        [Fact]
+        public void CopyOntoDirectory()
+        {
+            string testFile = GetTestFilePath();
+            using (File.Create(testFile))
             {
-                iCountErrors++;
-                printerr("Error_109uc! Incorrect exception thrown, exc==" + exc.ToString());
+                Assert.Throws<IOException>(() => Copy(testFile, Directory.GetCurrentDirectory()));
```

Comment:
I don't think this is testing what it's meant to be testing.  It's going to pass, but only because the resulting exception from trying to copy an open file is an IOException just as is the exception for copying onto a directory.